src/logging.cc

```diff
@@ -70,30 +70,67 @@ LogMessage::LogMessage(const char* file, int line, uint32_t level)
     path = path.substr(pos + 1, std::string::npos);
   }
 
+  // 'L' for log level
+  switch (gLogger_.LogFormat())
+  {
+  case Logger::Format::kDEFAULT: {
+    // LMMDD hh:mm:ss.ssssss
 #ifdef _WIN32
-  SYSTEMTIME system_time;
-  GetSystemTime(&system_time);
-  stream_ << level_name_[std::min(level, (uint32_t)Level::kINFO)]
-          << std::setfill('0') << std::setw(2) << (system_time.wMonth + 1)
-          << std::setw(2) << system_time.wDay << " " << std::setw(2)
-          << system_time.wHour << ':' << std::setw(2) << system_time.wMinute
-          << ':' << std::setw(2) << system_time.wSecond << "." << std::setw(6)
-          << system_time.wMilliseconds * 1000 << ' '
-          << static_cast<uint32_t>(GetCurrentProcessId()) << ' ' << path << ':'
-          << line << "] ";
+    SYSTEMTIME system_time;
+    GetSystemTime(&system_time);
+    stream_ << level_name_[std::min(level, (uint32_t)Level::kINFO)]
+            << std::setfill('0') << std::setw(2) << system_time.wMonth
+            << std::setw(2) << system_time.wDay << " " << std::setw(2)
+            << system_time.wHour << ':' << std::setw(2) << system_time.wMinute
+            << ':' << std::setw(2) << system_time.wSecond << "." << std::setw(6)
+            << system_time.wMilliseconds * 1000 << ' '
+            << static_cast<uint32_t>(GetCurrentProcessId()) << ' ' << path << ':'
+            << line << "] ";
 #else
-  struct timeval tv;
-  gettimeofday(&tv, NULL);
-  struct tm tm_time;
-  gmtime_r(((time_t*)&(tv.tv_sec)), &tm_time);
-  stream_ << level_name_[std::min(level, (uint32_t)Level::kINFO)]
-          << std::setfill('0') << std::setw(2) << (tm_time.tm_mon + 1)
-          << std::setw(2) << tm_time.tm_mday << " " << std::setw(2)
-          << tm_time.tm_hour << ':' << std::setw(2) << tm_time.tm_min << ':'
-          << std::setw(2) << tm_time.tm_sec << "." << std::setw(6) << tv.tv_usec
-          << ' ' << static_cast<uint32_t>(getpid()) << ' ' << path << ':'
-          << line << "] ";
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    struct tm tm_time;
+    gmtime_r(((time_t*)&(tv.tv_sec)), &tm_time);
+    stream_ << level_name_[std::min(level, (uint32_t)Level::kINFO)]
+            << std::setfill('0') << std::setw(2) << (tm_time.tm_mon + 1)
+            << std::setw(2) << tm_time.tm_mday << " " << std::setw(2)
+            << tm_time.tm_hour << ':' << std::setw(2) << tm_time.tm_min << ':'
+            << std::setw(2) << tm_time.tm_sec << "." << std::setw(6) << tv.tv_usec
+            << ' ' << static_cast<uint32_t>(getpid()) << ' ' << path << ':'
+            << line << "] ";
 #endif
+    break;
+  }
+  case Logger::Format::kISO8601: {
+    // YYYY-MM-DDThh:mm:ss L
```

Comment:
Maybe an addition of a `Z` specifier at the end to denote this is in UTC would be helpful, similar to Tanmay's comment.

> As format Z,  This date time has no offset from UTC (or 0 hour offset).

Reply:
@tanmayv25 I think I will use `Z` to highlight UTC time as described here. The doc for `gettimeofday` suggests that there can be complication when handling local time.
> The use of the timezone structure is obsolete; the tz argument
       should normally be specified as NULL.  (See NOTES below.)